libs/cpa/data-api/src/lib/modules/workshops/workshops.service.spec.ts

```diff
@@ -6,19 +6,78 @@ import { Repository } from 'typeorm';
 import { Workshop } from '@tamu-gisc/cpa/common/entities';
 
 import { WorkshopsService } from './workshops.service';
+import { throwError } from 'rxjs';
+
+jest.mock('../base/base.service');
 
 describe('WorkshopsService', () => {
-  let service: WorkshopsService;
+  let workshopsService: WorkshopsService;
+  let workshopsRepository: Repository<Response>;
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [WorkshopsService, { provide: getRepositoryToken(Workshop), useClass: Repository }]
+      providers: [
+        WorkshopsService,
+        {
+          provide: getRepositoryToken(Workshop),
+          useValue: {
+            findOne: jest.fn(),
+            update: jest.fn(),
+            delete: jest.fn()
+          }
+        }
+      ]
     }).compile();
 
-    service = module.get<WorkshopsService>(WorkshopsService);
+    workshopsService = module.get<WorkshopsService>(WorkshopsService);
+    workshopsRepository = module.get(getRepositoryToken(Workshop));
+  });
+
+  describe('Validation ', () => {
+    it('service should be defined', async () => {
+      expect(workshopsService).toBeDefined();
+    });
+  });
+
+  describe('addNewScenario', () => {
+    it('should return call repo.update, and should accept IResponseRequestPayload type as mock parameters ', async () => {
```

Comment:
Test descriptions, should describe a falsy/error return value

Consider using the `HttpStatus` enum from `@nestjs/common` to compare status codes instead of response strings.

Reply:
Done

Reply:
Left an unused import